### ISSUE TITLE
位置编码问题

### ISSUE THREAD

**Issue by wuxiaohui0** [yhcc/utcie#3]
@staticmethod
    def apply_rotary(x, sinusoidal_pos):
        sin, cos = sinusoidal_pos
        x1, x2 = x[..., 0::2], x[..., 1::2]
        # 如果是旋转query key的话，下面这个直接cat就行，因为要进行矩阵乘法，最终会在这个维度求和。（只要保持query和key的最后一个dim的每一个位置对应上就可以）
        # torch.cat([x1 * cos - x2 * sin, x2 * cos + x1 * sin], dim=-1)
        # 如果是旋转value的话，下面这个stack后再flatten才可以，因为训练好的模型最后一个dim是两两之间交替的。
        return torch.stack([x1 * cos - x2 * sin, x2 * cos + x1 * sin], dim=-1).flatten(-2, -1)
在这部分代码中是应用旋转位置编码，论文中提到的Triangle position embedding is incorporated
to mark the position of token pairs in the fea-ture map, indicating whether the cell is in the upper or lower triangles. It adds to S in Eq.(3)
before Attention.这个位置编码时在那里面实现的呢？

**Comment by yhcc** [comment]
看起来通过 https://github.com/yhcc/utcie/blob/414e70f23e1bbb1ca657a02e3548c827189aef7f/model/unify_model.py#L43-L58 这个地方的span_embed可以让模型感知是在哪个半区？
